### ISSUE TITLE
Not able to export bigger design

### ISSUE THREAD

**Issue by the3dfox** [penpot/penpot-exporter-figma-plugin#209]
Hi, after the rollback today I finally was able to export some designs. However, we are still unable to export our main design. When we try, at some point the browser freezes or the process is stuck at generating penpot file. The console log is the following, not sure if it is helpful:

```
VM534:7 Uncaught RangeError: Invalid string length
    at Object.push (<anonymous>:7:22416)
    at Bl (<anonymous>:7:17582)
    at $t (<anonymous>:7:17730)
    at cf (<anonymous>:7:20445)
    at am (<anonymous>:7:22720)
    at l.renderToStaticMarkup (<anonymous>:7:29462)
    at <anonymous>:266:5187
    at <anonymous>:238:85170
    at nn.next [as _nextOverride] (<anonymous>:13:21833)
    at nn.i [as _next] (<anonymous>:11:75107)
```

unfortunately, we can not share the figma file publicly but if you want to test it for yourself, we can invite you with your email address

**Comment by Cenadros** [comment]
Hi there @the3dfox. We're going to start working again in trying to solve the issues that are being reported and I've been looking at this one.

If you can invite us to the document with our mails (alejandro@runroom.com and jsala@runroom.com) to check the error it would be great.

Thank you in advance.

**Comment by BrianGilbert** [comment]
@Cenadros here is a design I've been trying to export, getting the following error:

```
VM257:136  Uncaught (in promise) RangeError: Maximum call stack size exceeded
    at xs (<anonymous>:136:14447)
    at gT (<anonymous>:136:39385)
    at ia (<anonymous>:136:44241)
    at Ru (<anonymous>:136:56519)
    at g_n (<anonymous>:238:30037)
    at IK (<anonymous>:238:29817)
    at tq (<anonymous>:238:32675)
    at l (<anonymous>:238:34897)
    at l (<anonymous>:238:35100)
    at l (<anonymous>:238:35100)
    ```

**Comment by the3dfox** [comment]
Sorry, took me some longer. Invited you today. Thanks for looking into it

**Comment by Cenadros** [comment]
Hi there @the3dfox,

I can confirm you that this issue is one of the most commons we're having right now. To give you some information this happens when there's a page too big to be handled by the current penpot library that handles the export in .svg. The .svg resulting from the export is so big that it reaches a size limit.

The penpot team is currently working in an alternative way of exporting these files and as soon as we can develop it in the plugin we'll address this issue.

Thank you for your help and patience.

**Comment by the3dfox** [comment]
Thanks for elaborating, glad to hear penpot team and you are actively working on this.

**Comment by fabiencharrasse** [comment]
> The penpot team is currently working in an alternative way of exporting these files and as soon as we can develop it in the plugin we'll address this issue.

Hi @Cenadros, can you provide an update on the progress?

**Comment by myfunnyandy** [comment]
Hi! We're sharing updates here https://github.com/penpot/penpot/issues/6358#issuecomment-2904208091

tl;dr: We're working on it, there's already good progress and we should have this fixed very soon. Thanks for the patience!
